config/database.php

```diff
@@ -144,6 +144,20 @@
 
 ];
 
+/*
+ * If the database host starts with a /, it is likely to be used with a
+ * unix socket. We can safely unset the host and the port since they
+ * likely won't be used in this configuration. This configurion likely
+ * won't be used in production, but rather when running with MAMP, XAMPP,
+ * WAMP, etc.
+ */
+
+if (substr($db['connections']['mysql']['host'], 0, 1) == '/') {
```

Comment:
IMO, it's safier just to add a new line in connections/mysql array, like this:
     'unix_socket' => env('UNIX_SOCKET', ''),
If the socket is set, laravel will use it first, so we don't need this config that depend on if the first character of "host" is a slash or not (which is dangerous)